pkg/cloud/awscloud/awscloud.go

```diff
@@ -624,9 +642,53 @@ func (a *AWS) Regions() ([]string, error) {
 	for _, r := range out.Regions {
 		result = append(result, aws.StringValue(r.RegionName))
 	}
+
+	return result, nil
+}
+
+func (a *AWS) Buckets() ([]string, error) {
+	out, err := a.s3.ListBuckets(nil)
+	if err != nil {
+		return nil, err
+	}
+
+	result := []string{}
+	for _, b := range out.Buckets {
+		result = append(result, aws.StringValue(b.Name))
+	}
+
 	return result, nil
 }
 
+func checkAWSPermissionMatrix(requiredPermission S3Permission, currentPermission S3Permission) bool {
+	requiredPermissions, exists := PermissionsMatrix[requiredPermission]
+	if !exists {
+		return false
+	}
+
+	for _, permission := range requiredPermissions {
+		if permission == currentPermission {
+			return true
+		}
+	}
+	return false
+}
+func (a *AWS) CheckBucketPermission(bucketName string, permission S3Permission) (bool, error) {
```

Comment:
Docstring would be nice explaning that it checks for "the requested permission or higher", so that the matrix is easier to understand.

Not super important though.

Reply:
I like good/short doc strings anyway